public/src/pages/StudentPage.tsx

```diff
@@ -184,58 +204,60 @@ export class StudentPage extends ViewPage {
         if (key === 0) {
             const coursesLinks: ILinkCollection[] = [];
             for (const course of this.activeUserCourses) {
-                const courseID = course.course.getId();
-                const allLinks: ILink[] = [];
-                allLinks.push({ name: "Labs" });
-                const labs = course.labs;
-                const gLabs: ILink[] = [];
-                labs.forEach((lab) => {
-                    if (lab.assignment.getIsgrouplab()) {
-                        gLabs.push({
-                            name: lab.assignment.getName(),
-                            uri: this.pagePath + "/courses/" + courseID + "/grouplab/" + lab.assignment.getId(),
-                        });
-                    } else {
-                        allLinks.push({
-                            name: lab.assignment.getName(),
-                            uri: this.pagePath + "/courses/" + courseID + "/lab/" + lab.assignment.getId(),
-                        });
-                    }
-                });
-                allLinks.push({ name: "Group Labs" });
-                allLinks.push(...gLabs);
-                allLinks.push({ name: "Repositories" });
-
-                const repos = await this.courseMan.getRepositories(
-                    courseID,
-                    [Repository.Type.USER,
-                    Repository.Type.GROUP,
-                    Repository.Type.COURSEINFO,
-                    Repository.Type.ASSIGNMENTS],
-                    );
-
-                allLinks.push({
-                    name: "User Repository", uri: repos.get(Repository.Type.USER), absolute: true,
-                });
-
-                allLinks.push({
-                    name: "Group Repository", uri: repos.get(Repository.Type.GROUP), absolute: true,
-                });
-
-                allLinks.push({
-                    name: "Course Info", uri: repos.get(Repository.Type.COURSEINFO), absolute: true,
-                });
-                allLinks.push({
-                    name: "Assignments", uri: repos.get(Repository.Type.ASSIGNMENTS), absolute: true,
-                });
-
-                allLinks.push({
-                    name: "New Group", uri: this.pagePath + "/courses/" + courseID + "/members",
-                });
-                coursesLinks.push({
-                    item: { name: course.course.getCode(), uri: this.pagePath + "/courses/" + courseID },
-                    children: allLinks,
-                });
+                if (course.enrollment.getState() !== Enrollment.DisplayState.ARCHIVED) {
```

Comment:
This code is a bit long; should be split into one or two methods.
Also, it seems a bit strange to need to check if a course is archived when iterating over the active courses.

Reply:
I guess, "active" here refers to the fact that only the courses where enrollment status in not pending are being included here.
Btw, are names like "archived"/"active" really that suitable when we talk about their presence in the user's page menus? Would it be better to rename them to something like "hidden" and, not sure here, "visible", or something else?

Reply:
Refactored out the repo part of the long method.

Reply:
Show and Hide are good names in this context.

Reply:
What about `VisibilityState` enum of `Enrollment`? They are called `Active`, `Archived`, `Favorite` now. Let them be, or rename to `Visible`, `Hidden`, `Favorite`?

Reply:
I like it

Reply:
Done.